weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx

```diff
@@ -0,0 +1,315 @@
+  };
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popper' : undefined;
+
+  // Call download query
+  const {useCallsExport} = useWFHooks();
+  const download = useCallsExport();
+  const {sortBy, lowLevelFilter, filterBy} = useFilterSortby(
+    callQueryParams.filter,
+    callQueryParams.gridFilter,
+    callQueryParams.gridSort
+  );
+
+  const onClickDownload = (contentType: ContentType) => {
+    lowLevelFilter.callIds = selectionState.selectedChecked
+      ? selectedCalls
+      : undefined;
+    // TODO(gst): allow user to specify offset?
+    const offset = 0;
+    const limit = selectionState.limitChecked
+      ? Math.min(MAX_EXPORT, parseInt(selectionState.limit ?? '100', 10))
+      : MAX_EXPORT;
+    download(
+      callQueryParams.entity,
+      callQueryParams.project,
+      contentType,
+      lowLevelFilter,
+      limit,
+      offset,
+      sortBy,
+      filterBy,
+      visibleColumns
+    ).then(blob => {
+      const fileExtension = fileExtensions[contentType];
+      const date = new Date().toISOString().split('T')[0];
+      const fileName = `weave_export_${callQueryParams.project}_${date}.${fileExtension}`;
+      initiateDownloadFromBlob(blob, fileName);
+      setAnchorEl(null);
+    });
+    setSelectionState({allChecked: true});
+  };
+
+  return (
+    <>
+      <span ref={ref}>
+        <Button
+          className={rightmostButton ? 'mr-16' : 'mr-4'}
+          icon="export-share-upload"
+          variant="ghost"
+          onClick={onClick}
+          disabled={disabled}
+        />
+      </span>
+      <Popover
+        id={id}
+        open={open}
+        anchorEl={anchorEl}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'center',
+        }}
+        onClose={() => {
+          setAnchorEl(null);
+          setSelectionState({allChecked: true});
+        }}>
+        <Tailwind>
+          <div className="min-w-[460px] p-12">
+            <div className="flex items-center pb-8">
+              <div className="flex-auto text-xl font-semibold">Export</div>
+              <div className="ml-16 text-moon-500">
+                {maybePluralize(visibleColumns.length, 'column', 's')}
+              </div>
+            </div>
+            <SelectionCheckboxes
+              selectedCalls={selectedCalls}
+              numTotalCalls={numTotalCalls}
+              selectionState={selectionState}
+              setSelectionState={setSelectionState}
+            />
+            <DownloadGrid onClickDownload={onClickDownload} />
+            <div className="mt-12 flex items-center text-moon-500">
+              <IconOnlyPill color="moon" icon="warning" />
+              <div className="ml-6">
+                This export experience is in beta and subject to change.
+              </div>
+            </div>
+          </div>
+        </Tailwind>
+      </Popover>
+    </>
+  );
+};
+
+const SelectionCheckboxes: FC<{
+  selectedCalls: string[];
```

Comment:
Seems like this could just be numSelected, you aren't using anything other than its length.